resource/stack/stack.js

```diff
@@ -77,54 +69,33 @@ const upgradeStack = Promise.coroutine(function* ( slack, envelope ) {
 	}
 	
 	//get the environments
-	const storedEnvironments = yield environment.getAll().catch( () => undefined );
-	if ( storedEnvironments === undefined ) { 
-		return {status: 500, data: {message: "Unable to get enviorments from the database"}}; 
-	}
-
-	
-	//get the rancher data
-	const envRequests = [];
-	for ( let i = 0; i < storedEnvironments.length; i++ ) {
-		envRequests.push( rancherFn( storedEnvironments[i].baseUrl, {
-			key: storedEnvironments[i].key,
-			secret: storedEnvironments[i].secret
-		} ).catch(() => console.error("Authorization failed for ", storedEnvironments[i].name)).then(Promise.coroutine(function*(rancher) {
-			if (rancher === undefined) {
-				throw new Error("Authorization failed for " + storedEnvironments[i].name);
-			}
-			const environments = yield rancher.listEnvironments();
-			return environments[_.keys( environments )[0]];
-		})));
+	const enviorments = yield rancherFn(rancherUrl, rancherUser).then(elm => elm.listEnvironments()).catch(() => undefined);
+	if ( enviorments === undefined ) { 
```

Comment:
the code should handle situations where the request went through, but the user did not have access to any rancher environments (e.g. empty array).